src/psl.cpp

```diff
@@ -0,0 +1,180 @@
+#include <algorithm>
+#include <fstream>
+#include <iostream>
+#include <string>
+
+#include "psl.h"
+#include "punycode.h"
+
+namespace Url
+{
+
+    PSL::PSL(std::istream& stream)
+    {
+        std::string line;
+        while (std::getline(stream, line))
+        {
+            // Skip blank lines
+            if (!line.size())
+            {
+                continue;
+            }
+
+            // Skip comments
+            if (line.size() >= 2 && line[0] == '/' && line[1] == '/')
```

Comment:
There should actually be another rule to set the line to be only up to the first whitespace.

But TIL about `int compare (size_t pos, size_t len, const string& str) const;`